src/engines/gstenginepipeline.cpp

```diff
@@ -153,17 +153,28 @@ bool GstEnginePipeline::ReplaceDecodeBin(GstElement* new_bin) {
 }
 
 bool GstEnginePipeline::ReplaceDecodeBin(const QUrl& url) {
-  GstElement* new_bin = nullptr;
+  GstElement* new_bin = CreateDecodeBinFromUrl(url);
+  return ReplaceDecodeBin(new_bin);
+}
 
+GstElement* GstEnginePipeline::CreateDecodeBinFromUrl(const QUrl& url) {
+  GstElement* new_bin = nullptr;
 #ifdef HAVE_SPOTIFY
   if (url.scheme() == "spotify") {
     new_bin = gst_bin_new("spotify_bin");
+    if (!new_bin) return NULL; 
```

Comment:
Use `nullptr` instead of `NULL` everywhere.

Reply:
Fixed